### PR TITLE
Add ExampleLength cop

### PR DESCRIPTION
**Pull request by andyw8**
This was suggested by @agrimm in issue #40

A long example is a form of the Obscure Test smell:
http://xunitpatterns.com/Obscure%20Test.html

I've chosen a default maximum of 5 lines on the basis that a typical
example would contain 3 lines of setup, one line of execution and one
line of assertion. However, this is quite dependent on the style used in
the codebase. Suites which rely more on `let` and `before` blocks would
likely have shorter example blocks.

Somewhat ironically, I had to set the configuration for rubocop-rspec
itself to quite a high value (12 lines).

### CODE REVIEW

File: lib/rubocop/cop/rspec/long_example.rb [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+          add_offense(node, :expression, message(length, max))
+        end
+
+        private
+
+        def max
+          cop_config['Max']
+        end
+
+        def code_length(node)
```
Comment (andyw8) [inline lib/rubocop/cop/rspec/long_example.rb]: Based on https://github.com/bbatsov/rubocop/blob/master/lib/rubocop/cop/metrics/method_length.rb

File: lib/rubocop/cop/rspec/long_example.rb [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+      #     expect(result).to be(true)
+      #   end
+      #
+      #   # good
+      #   it do
+      #     service = described_class.new
+      #     result = service.call
+      #     expect(result).to be(true)
+      #   end
+      class LongExample < Cop
```
Comment (andyw8) [inline lib/rubocop/cop/rspec/long_example.rb]: Maybe this should be called `ExampleLength` for consistency with RuboCop?

**Comment by andyw8** [comment]
TODO:
- [ ] Add changelog entry

File: lib/rubocop/cop/rspec/long_example.rb (hunk 2 above)
Comment (mikegee) [inline lib/rubocop/cop/rspec/long_example.rb]: That seems like a good idea.

Comment (nijikon) [inline lib/rubocop/cop/rspec/long_example.rb]: Yes, let's rename.

Comment (andyw8) [inline lib/rubocop/cop/rspec/long_example.rb]: @nijikon done.

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,7 @@
 
 ## master (unreleased)
 
+* Add `RSpec::LongExample` cop. ([@andyw8][])
```
Comment (nijikon) [inline CHANGELOG.md]: Rspec::ExampleLength

File: spec/rubocop/cop/rspec/example_length_spec.rb [hunk 4]
```diff
@@ -0,0 +1,63 @@
+# encoding: utf-8
+
+describe RuboCop::Cop::RSpec::ExampleLength, :config do
+  subject(:cop) { described_class.new(config) }
+  let(:cop_config) do
```
Comment (nijikon) [inline spec/rubocop/cop/rspec/example_length_spec.rb]: In one line.

File: spec/rubocop/cop/rspec/example_length_spec.rb [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    {
+      'Max' => 3
+    }
+  end
+
+  it 'allows an empty example' do
+    inspect_source(cop, ['it do',
```
Comment (nijikon) [inline spec/rubocop/cop/rspec/example_length_spec.rb]: Format inspect_source like in other examples, see https://github.com/nevir/rubocop-rspec/commit/bd5fe5997de2d716755033b12df89cb66d9731c2 for reference

**Comment by andyw8** [comment]
@nijikon updated based on your suggestions.

**Comment by nijikon** [comment]
@andyw8 please rebase on top of master

**Comment by andyw8** [comment]
@nijikon done

**Pull request merged**